packages/ckeditor5-ui/src/panel/balloon/balloonpanelview.ts

```diff
@@ -387,6 +404,27 @@ export default class BalloonPanelView extends View {
 		this.listenTo( global.window, 'resize', () => {
 			this.attachTo( options );
 		} );
+
+		// Hide the panel if the target element is no longer visible.
+		if ( targetElement && !this._resizeObserver ) {
+			const checkVisibility = () => {
+				// If the target element is no longer visible, hide the panel.
+				if ( !isVisible( targetElement ) ) {
+					this.unpin();
+				}
+			};
+
+			// Element is being resized to 0x0 after it's parent became hidden,
+			// so we need to check size in order to determine if it's visible or not.
+			this._resizeObserver = new ResizeObserver( targetElement, checkVisibility );
```

Comment:
This code will likely be obsolete after this fix. Consider removing it https://github.com/ckeditor/ckeditor5/blob/ea0947ab56e3b6923b989cd80606688524fff474/packages/ckeditor5-ui/src/tooltipmanager.ts#L431-L437.

Reply:
@oleq I tried, but it's not that simple and makes tooltips buggy. It turns out that we cannot bind to `isVisible` of balloon panel view because it becomes hidden during updating of its position (and then it became is visible again).